rustls/src/msgs/message.rs

```diff
@@ -214,6 +190,66 @@ impl OpaqueMessage {
     pub const MAX_WIRE_SIZE: usize = (Self::MAX_PAYLOAD + Self::HEADER_SIZE) as usize;
 }
 
+/// A borrowed version of [`OpaqueMessage`].
+pub struct BorrowedOpaqueMessage<'a> {
+    pub typ: ContentType,
+    pub version: ProtocolVersion,
+    pub payload: BorrowedPayload<'a>,
+}
+
+#[allow(dead_code)]
+impl<'a> BorrowedOpaqueMessage<'a> {
+    pub(crate) fn read(r: &mut ReaderMut<'a>) -> Result<Self, MessageError> {
+        let (typ, version, len) = r.as_reader(read_opaque_message_minus_payload)?;
+
+        let mut sub = r
+            .sub(len as usize)
+            .map_err(|_| MessageError::TooShortForLength)?;
+        let payload = BorrowedPayload::read(&mut sub);
+
+        Ok(Self {
+            typ,
+            version,
+            payload,
+        })
+    }
+}
+
+fn read_opaque_message_minus_payload(
```

Comment:
Naming nit: I'd probably call this `read_opaque_message_header()`.

Reply:
Renamed